src/index.js

```diff
@@ -13,4 +14,21 @@ root.render(
   </React.StrictMode>
 );
 
-reportWebVitals();
+// if ("serviceWorker" in navigator) {
+//   navigator.serviceWorker.register("service-worker.js").then(registration => {
+//     console.log("it worked", registration)
+//   }).catch(error => { console.log("fail",error) });
+// }
+
+if ('serviceWorker' in navigator) {
+  navigator.serviceWorker.register('../service-worker.js').then(function(reg) {
```

Comment:
Let's remove the directory definition (../) before `service-worker.js` & move the file (service-worker.js) into `public` directory. So create-react-app will able to see the file accordingly. Once you are done with it, you will get an error sth. like the following:

```
Uncaught SyntaxError: Unexpected token 'export' (at service-worker.js:23:1)
```

That's because create-react-app doesn't recognise ES6-native export definitions within the service worker file. So you can revisit the service worker file by the example here & update the file like how it has been mentioned over here in this article: [What Are Service Workers and How to Use Them?](https://rajat-m.medium.com/what-are-service-workers-and-how-to-use-them-e993c1f497e6 ) - a better & cleaner versions which is pretty much enough for our purposes within this feature 😉